src/Microsoft.AspNet.Razor.Runtime/TagHelpers/HtmlElementNameAttribute.cs

```diff
@@ -45,6 +47,29 @@ public HtmlElementNameAttribute([NotNull] string tag, [NotNull] params string[]
         /// <summary>
         /// An <see cref="IEnumerable{string}"/> of tag names for the <see cref="TagHelper"/> to target.
         /// </summary>
-        public IEnumerable<string> Tags { get; private set; }
+        public IEnumerable<string> Tags
+        {
+            get
+            {
+                return _tags;
+            }
+            private set
+            {
+                foreach (var tagName in value)
+                {
+                    ValidateTagName(tagName);
+                }
+
+                _tags = value;
+            }
+        }
+
+        private static void ValidateTagName(string tagName)
```

Comment:
- handle `null` names here too; allow `params` constructor to run through its parameters once
- not that big a deal that this means you'll validate `tag` parameters are non-`null`; `[NotNull]` will eventually prevent that case